### PR TITLE
Generate Shoot template in Provisioner

### PR DESCRIPTION
**Pull request by Szymongib**
**Description**

Changes proposed in this pull request:

- Generate Shoot template
- Render template with values

**Related issue(s)**

### CODE REVIEW

File: components/provisioner/internal/templates/render.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+func RenderTemplate(text string, values Values) ([]byte, error) {
+	templ, err := template.New("").Parse(text)
+	if err != nil {
+		return nil, fmt.Errorf("error while parsing renderTemplate: %s", err.Error())
+	}
+
+	buffer := new(bytes.Buffer)
+	err = templ.Execute(buffer, values)
+	if err != nil {
+		return nil, fmt.Errorf("error while rendering renderTemplate: %s", err.Error())
```
Comment (rafalpotempa) [inline components/provisioner/internal/templates/render.go:25]: I'm not sure about meaning of this error message. Did you mean (`error while rendering template '%s'`, text)
```suggestion
		return nil, fmt.Errorf("error while rendering template '%s': %s", text, err.Error())
```
?

File: components/provisioner/templates/cluster-bom.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+apiVersion: "hub.k8s.sap.com/v1"
+kind: ClusterBom
+metadata:
+  name: {{ .ShootName }}
+spec:
+  secretRef: {{ .ShootName }}.kubeconfig
+  applicationConfigs:
+  - id: kyma
+    configType: helm
+    readyRequirements:
```
Comment (rafalpotempa) [inline components/provisioner/templates/cluster-bom.yaml:10]: OIDC config could go here

File: components/provisioner/templates/shoot.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  region: '{{ .Region }}'
+  secretBindingName: '{{ .GardenerSecretName }}'
+status:
+  gardener:
+    id: ""
+    name: ""
+    version: ""
+  hibernated: false
+  technicalID: ""
+  uid: ""
```
Comment (rafalpotempa) [inline components/provisioner/templates/shoot.yaml:65]: How do you think: should we use the full possible shoot config, or just the pure minimum?

Comment (Szymongib) [reply components/provisioner/templates/shoot.yaml:65]: I think we should keep it as close as possible to the Shoot created by the Provisioner.

File: components/provisioner/templates/cluster-bom.yaml (hunk 2 above)
Comment (Szymongib) [reply components/provisioner/templates/cluster-bom.yaml:10]: Yes, I am waiting to confirm the configuration.

File: components/provisioner/internal/templates/render.go (hunk 1 above)
Comment (Szymongib) [reply components/provisioner/internal/templates/render.go:25]: Good catch. Did a little too much renaming, I will fix it with the next changes.

**Pull request closed without merging**